### PR TITLE
Add another output (optional) for the final cell state for LSTM.

### PR DESCRIPTION
**Pull request by yuanbyu**
There are useful LSTM models that actually use the final cell state.

### CODE REVIEW

File: onnx/defs/rnn/defs.cc [hunk 1]
```diff
@@ -235,6 +235,7 @@ Equations (Default: f=Sigmoid, g=Tanh):
 
 
 OPERATOR_SCHEMA(LSTM)
+    .SinceVersion(2)
```
Comment (linkerzhang) [inline onnx/defs/rnn/defs.cc]: When checking in this change, kindly remember to bump the global version in schema.h accordingly please. The version here may be updated too if the global version is changed by other users. For example, PR #390 .

Comment (dzhulgakov) [reply onnx/defs/rnn/defs.cc]: Yep, please do this

File: onnx/defs/rnn/defs.cc [hunk 2]
```diff
@@ -54,7 +55,6 @@ std::function<void(OpSchema&)> RNNDocGenerator(const char* name) {
     };
```
Comment (dzhulgakov) [inline onnx/defs/rnn/defs.cc]: I wonder whether we should make output Y_h optional too for symmetry purposes. You can probably sneak it in without bumping the version as it's just an optional output and optionals are not yet properly handled everywhere anyway.

Comment (yuanbyu) [reply onnx/defs/rnn/defs.cc]: I am fine with this. Should we go even further to make the outputs of every op that has >1 outputs optional?

**Comment by yuanbyu** [comment]
OK. I made every output of LSTM optional. PTAL.

File: onnx/defs/rnn/old.cc [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+  - ot = f(Xt*(Wo^T) + Ht-1*Ro + Po (.) Ct + Wbo + Rbo)
+
+  - Ht = ot (.) h(Ct)
+)DOC")
+    .Attr("activations", "A list of 3 (or 6 if bidirectional) activation functions "
+          "for input, output, forget, cell, and hidden. The activation functions must "
+          "be one of the activation functions specified above. Optional: See the equations "
+          "for default if not specified.",
+          AttributeProto::STRINGS)
```
Comment (bddppq) [inline onnx/defs/rnn/old.cc]: This was optional.

**Comment by linkerzhang** [comment]
With latest clarification and discussion, checking #410 for more details please. This is not a breaking change, no need bumping version please. Thank you!

File: onnx/defs/rnn/old.cc (hunk 3 above)
Comment (linkerzhang) [reply onnx/defs/rnn/old.cc]: Per latest discussion and decision, this change does not need to bump version, so revert all changes in old.cc now.

**Comment by linkerzhang** [comment]
All the comments were resolved. It's now going to be checked-in.

**Pull request merged**